### DISCUSSION TITLE
Documentation: raw helper "value can be empty array"

### DISCUSSION THREAD

**Discussion by leonardbinet** [mikro-orm/mikro-orm#6247] [Q&A] [answered]
In https://mikro-orm.io/docs/raw-queries#raw-helper, we find:

```
// value can be empty array
await em.find(User, { [raw('(select 1 = 1)')]: [] });
```

I fail to understand what the empty array `[]` means. Is it a special syntax to indicate to the orm that only the key should be incorporated?

More generally, examples in this section show examples using `=` operator, can I use `raw` helper to build queries with other operators, for instance `IN`, `LIKE` etc ?

In my case, I would like to perform a query on a `jsonb` column named `input_json`, such as:

```
 select "m0".* from "mapping_row" as "m0" where  "m0"."input_json" in ('{"1":"lyon"}', '{"1":"paris"}');
```

Failing to do so in a "regular" way, I would like to try to build such query using the `raw` helper, but I fail to fully grasp how it is supposed to be used.

Relates to https://github.com/mikro-orm/mikro-orm/issues/6246

Thx for your work on this library 🙏

**Comment by B4nan** [comment, marked as answer]
This way you are able to pass a raw fragment without any value. Normally you would use the key for left hand side and the value to right hand size, as in `[raw('foo')]: 123` resulting in `foo = 1` or something similar. If you pass empty array there, it would be just `foo` without the equals sign.

> More generally, examples in this section show examples using = operator

There are examples for operators too, like here: https://mikro-orm.io/docs/raw-queries#raw-fragments-in-filters

**Reply by leonardbinet** [reply]
ok, this wasn't very clear to me, makes sense, I managed to perform my query using it 👍

**Reply by B4nan** [reply]
I'll try to improve the docs a bit around this when I have some spare time. Maybe next week, I am really toasted now and the week is just in the middle 🙃
